secondary.tf

```diff
@@ -0,0 +1,38 @@
+resource "aws_s3_bucket" "secondary" {
+  provider = aws.secondary
+
+  bucket = var.secondary_name
+
+  force_destroy = var.force_destroy
+
+  tags = var.tags
+}
+
+resource "aws_s3_bucket_versioning" "secondary" {
+  provider = aws.secondary
+
+  bucket = aws_s3_bucket.secondary.bucket
+
+  versioning_configuration {
+    status = "Enabled"
+  }
+}
+
+resource "aws_s3_bucket_acl" "secondary" {
```

Comment:
`provider` is missing... however, see also my comment that I'd prefer to omit this resource in a public module.